src/network/clientpackethandler.cpp

```diff
@@ -1384,26 +1385,31 @@ void Client::handleCommand_ModChannelSignal(NetworkPacket *pkt)
 			infostream << "Server refused our mod channel leave on channel `" << channel
 				<< "`" << std::endl;
 			break;
-#ifndef NDEBUG
 		case MODCHANNEL_SIGNAL_CHANNEL_NOT_REGISTERED:
 			// Generally unused, but ensure we don't do an implementation error
 			infostream << "Server tells us we sent a message on channel `" << channel
 				<< "` but we are not registered. Message was dropped." << std::endl;
 			break;
 #endif
-		case MODCHANNEL_SIGNAL_SET_READ_ONLY:
-			m_modchannel_mgr->setChannelState(channel, MODCHANNEL_STATE_READ_ONLY);
+		case MODCHANNEL_SIGNAL_SET_STATE:
+			u8 state;
```

Comment:
Needs brackets around this case for proper variable deconstruction within this part.